internal/http/routes/bookmark.go

```diff
@@ -1,118 +1,189 @@
 package routes
 
 import (
+	"fmt"
+	"html/template"
 	"net/http"
 	"strconv"
-
-	fp "path/filepath"
+	"strings"
 
 	"github.com/gin-gonic/gin"
-	"github.com/go-shiori/shiori/internal/config"
+	"github.com/go-shiori/shiori/internal/dependencies"
 	"github.com/go-shiori/shiori/internal/http/context"
 	"github.com/go-shiori/shiori/internal/http/response"
 	"github.com/go-shiori/shiori/internal/model"
-	ws "github.com/go-shiori/shiori/internal/webserver"
+	"github.com/gofrs/uuid/v5"
 	"github.com/sirupsen/logrus"
 )
 
 type BookmarkRoutes struct {
 	logger *logrus.Logger
-	deps   *config.Dependencies
+	deps   *dependencies.Dependencies
+}
+
+func NewBookmarkRoutes(logger *logrus.Logger, deps *dependencies.Dependencies) *BookmarkRoutes {
+	return &BookmarkRoutes{
+		logger: logger,
+		deps:   deps,
+	}
 }
 
 func (r *BookmarkRoutes) Setup(group *gin.RouterGroup) model.Routes {
-	//group.GET("/:id/archive", r.bookmarkArchiveHandler)
-	//group.GET("/:id/content", r.bookmarkContentHandler)
+	group.GET("/:id/archive", r.bookmarkArchiveHandler)
+	group.GET("/:id/archive/file/*filepath", r.bookmarkArchiveFileHandler)
+	group.GET("/:id/content", r.bookmarkContentHandler)
+	group.GET("/:id/thumb", r.bookmarkThumbnailHandler)
 	group.GET("/:id/ebook", r.bookmarkEbookHandler)
+
 	return r
 }
 
-// func (r *BookmarkRoutes) bookmarkContentHandler(c *gin.Context) {
-// 	ctx := context.NewContextFromGin(c)
+func (r *BookmarkRoutes) getBookmark(c *context.Context) (*model.BookmarkDTO, error) {
+	bookmarkIDParam, present := c.Params.Get("id")
+	if !present {
+		response.SendError(c.Context, http.StatusBadRequest, "Invalid bookmark ID")
+		return nil, model.ErrBookmarkInvalidID
+	}
+
+	bookmarkID, err := strconv.Atoi(bookmarkIDParam)
+	if err != nil {
+		r.logger.WithError(err).Error("error parsing bookmark ID parameter")
+		response.SendInternalServerError(c.Context)
+		return nil, err
+	}
+
+	if bookmarkID == 0 {
+		response.SendError(c.Context, http.StatusNotFound, nil)
+		return nil, model.ErrBookmarkNotFound
+	}
```

Comment:
I think (because this is legacy) this is here because the default value for an `int` is `0`, so if for any reason we end up with the default value for an int, we show an appropriate error for that and help pinpoint better where the issue is. I think we could remove it because we are never going to find a `id=0` bookmark, but left it because it doesn't involve too many likes of code.